src/btf.cpp

```diff
@@ -38,151 +39,116 @@ static __u32 type_cnt(const struct btf *btf)
 #endif
 }
 
-static unsigned char *get_data(const char *file, ssize_t *sizep)
+__s32 BTF::start_id(const struct btf *btf) const
 {
-  struct stat st;
-
-  if (stat(file, &st))
-    return nullptr;
-
-  FILE *f;
-
-  f = fopen(file, "rb");
-  if (!f)
-    return nullptr;
+  return btf == vmlinux_btf ? 1 : vmlinux_btf_size + 1;
+}
 
-  unsigned char *data;
-  unsigned int size;
+static int libbpf_print(enum libbpf_print_level level,
+                        const char *msg,
+                        va_list ap)
+{
+  fprintf(stderr, "BTF: (%d) ", level);
+  return vfprintf(stderr, msg, ap);
+}
 
-  size = st.st_size;
+BTF::BTF(const std::set<std::string> &modules) : state(NODATA)
+{
+  if (bt_debug != DebugLevel::kNone)
+    libbpf_set_print(libbpf_print);
```

Comment:
This belongs at the main.cpp level, not in here. If I use btf.cpp in a library, I shouldn't get libbpf behavior altered from under me.